apps/emqx_machine/src/emqx_machine_boot.erl

```diff
@@ -135,6 +135,12 @@ add_app(G, App, undefined) ->
     ?SLOG(debug, #{msg => "app_is_not_loaded", app => App}),
     %% not loaded
     add_app(G, App, []);
+% We ALWAYS want to add `emqx_conf', even if no other app declare a
+% dependency on it.  Otherwise, emqx may fail to load the config
+% schemas, especially in the test profile.
+add_app(G, App = emqx_conf, []) ->
```

Comment:
HAHA, Good job!

I found out by accident that adding hocon to emqx_conf's include_application is the only way to get it to start before emqx. Otherwise it would fail to start.

Reply:
Indeed!  And, somehow, it only behaves like that in a release (getting inserted into `applications` by transitivity by Rebar, maybe?).  In the tests, I discovered that this does not happen, and the tests rely on it not starting!